### DISCUSSION TITLE
build error in docker

### DISCUSSION THREAD

**Discussion by chinawch007** [StarRocks/starrocks#4155] [Announcements]
I follow this.
> docker run -it --name sr -d starrocks/dev-env:main
docker exec -it sr /bin/bash
git clone https://github.com/StarRocks/starrocks.git
cd starrocks
sh build.sh   

And I meet this error:
>Build Frontend Modules: fe-common,spark-dpp,fe-core
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for com.starrocks:spark-dpp:jar:1.0.0
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] fe-common                                                          [jar]
[INFO] spark-dpp                                                          [jar]
[INFO] fe-core                                                            [jar]
[INFO] 
[INFO] ----------------------< com.starrocks:fe-common >-----------------------
[INFO] Building fe-common 1.0.0                                           [1/3]
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from spring-plugins: https://repo.spring.io/plugins-release/org/apache/maven/plugins/maven-checkstyle-plugin/3.1.1/maven-checkstyle-plugin-3.1.1.pom
Downloading from cloudera-public: https://repository.cloudera.com/artifactory/public/org/apache/maven/plugins/maven-checkstyle-plugin/3.1.1/maven-checkstyle-plugin-3.1.1.pom
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-checkstyle-plugin/3.1.1/maven-checkstyle-plugin-3.1.1.pom
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] fe-common 1.0.0 .................................... FAILURE [  6.457 s]
[INFO] spark-dpp 1.0.0 .................................... SKIPPED
[INFO] fe-core 3.4.0 ...................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.564 s
[INFO] Finished at: 2022-03-16T02:39:36Z
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.maven.plugins:maven-checkstyle-plugin:jar:3.1.1: Could not transfer artifact org.apache.maven.plugins:maven-checkstyle-plugin:pom:3.1.1 from/to spring-plugins (https://repo.spring.io/plugins-release/): Transfer failed for https://repo.spring.io/plugins-release/org/apache/maven/plugins/maven-checkstyle-plugin/3.1.1/maven-checkstyle-plugin-3.1.1.pom: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException

What should I do? Please help me.

**Comment by zombee0** [comment]
I encountered the same problem, the reason is that the java's cacerts is not good, i fixed it by downloading the newest jdk and replaced the old one.
